core/amber/src/main/scala/edu/uci/ics/backenderror/Error.scala

```diff
@@ -0,0 +1,13 @@
+package edu.uci.ics.backenderror
+
+/**
+  * @param errorName a descriptive name of the error
+  * @param errorSource where the error is occurring. eg: "Engine:Controller:CreateWorklow"
+  * @param errorAdditionalParams details about the error: is this an unexpected exception or a constraint-violation, stacktrace etc.
+  */
+case class Error(errorName: String, errorSource:String, errorAdditionalParams: Map[String, String]) {
```

Comment:
`errorMessage` to replace `errorName`? because it seems `errorName` is used as message

Reply:
just to echo another comment, maybe `WorkflowExecutionError` to replace `Error`

Reply:
done the other changes. I am not calling it WorkflowExecutionError for now, because it is quite general and can be used for other purposes